extensions/notebook/src/book/bookTrustManager.ts

```diff
@@ -0,0 +1,161 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as constants from './../common/constants';
+import { BookTreeItem } from './bookTreeItem';
+import { BookModel } from './bookModel';
+
+export interface IBookTrustManager {
+	isNotebookTrustedByDefault(notebookUri: string): boolean;
+	setBookAsTrusted(bookRootPath: string): boolean;
+	setBookAsUntrusted(bookRootPath: string): boolean;
+}
+
+export interface IBookTrustManagerWorkspaceDetails {
+	workspaceFolders: vscode.WorkspaceFolder[];
+	getConfiguration(section?: string | undefined): vscode.WorkspaceConfiguration;
+}
+
+enum TrustBookOperation {
+	Add,
+	Remove
+}
+
+export class BookTrustManager implements IBookTrustManager {
+
+	private static notebookConfiguration: string = constants.notebookConfigKey;
+	private static notebookTrustedBooksConfiguration: string = constants.trustedBooksConfigKey;
+	private trustedLocalBooks: string[] = [];
+
+	constructor(private books: BookModel[], private workspaceDetails?: IBookTrustManagerWorkspaceDetails) {
```

Comment:
I'm not really a fan of how we're passing in the list of books and then just assuming that that specific array is going to always have the latest set. There should really be a BookManagerService or something that does this that things like this can query for the current set of books and whatnot. I know this is adding onto an already largish set of changes but this is going to start getting messy fast if we keep adding things like this that do their own thing.

@chlafreniere @MaddyDev Thoughts on this?

Reply:
# resolved